lib/Fhp/BaseAction.php

```diff
@@ -75,13 +80,13 @@ public function serialize(): string
         if (!$this->needsTan()) {
             throw new \RuntimeException('Cannot serialize this action, because it is not waiting for a TAN.');
         }
-        return serialize([$this->requestSegmentNumbers, $this->tanRequest, $this->paginationToken]);
+        return serialize([$this->requestSegmentNumbers, $this->tanRequest, $this->paginationToken, $this->message]);
```

Comment:
Könnte es passieren, dass in `$this->message` persönliche Daten drin sind? Der Login hat ja mehrere Request-Segmente und deine Implementierung in `FinTs` verhindert deswegen derzeit, dass `setMessage()` dafür aufgerufen wird. Aber darauf sollte man sich lieber nicht verlassen. Z.B. wenn wir den Geschäftsvorfall "PIN ändern" implementieren, den man separat _nach_ dem Login ausführt und der vermutlich eine TAN erfordert, dann wollen wir nicht, dass der neue PIN hier serialisiert wird.

Vielleicht können wir eine `PaginateableAction extends BaseAction` machen, wo dieses Extra und das `Rueckmeldungscode::PAGINATION`-Lesen implementiert sind. Und Subklassen davon müssen dann garantieren, dass ihre Requests sauber sind.

Reply:
Ja dann bin ich auch für eine PaginateableAction, da sollten keine persönlichen Daten drin sein.